flusher.go

```diff
@@ -397,33 +404,40 @@ func (s *Server) flushForward(wms []WorkerMetrics) {
 			}
 			// the exporter doesn't know that these two are "different"
 			jm.Type = "timer"
-			jsonMetrics = append(jsonMetrics, jm)
+			dest, _ := s.ForwardDestinations.Get(jm.MetricKey.String())
+			jsonMetricsByDestination[dest] = append(jsonMetricsByDestination[dest], jm)
 		}
 	}
 	s.statsd.TimeInMilliseconds("forward.duration_ns", float64(time.Since(exportStart).Nanoseconds()), []string{"part:export"}, 1.0)
 
-	s.statsd.Gauge("forward.post_metrics_total", float64(len(jsonMetrics)), nil, 1.0)
-	if len(jsonMetrics) == 0 {
-		log.Debug("Nothing to forward, skipping.")
-		return
-	}
+	metrics_total := 0
+	for dest, batch := range jsonMetricsByDestination {
+		if len(batch) == 0 {
+			log.Debug("Nothing to forward, skipping.")
```

Comment:
is it worth throwing in the dest here? it might be helpful to know if some host is consistently getting skipped.